lib/clientv2/globalnetworkpolicy.go

```diff
@@ -113,3 +147,21 @@ func defaultPolicyTypesField(spec *apiv2.PolicySpec) {
 		}
 	}
 }
+
+func removePrefixFromGlobalPolicies(res *apiv2.GlobalNetworkPolicyList) {
+	for i, _ := range res.Items {
+		removePrefixFromGlobalPolicy(&res.Items[i])
+	}
+}
+
+func removePrefixFromGlobalPolicy(policy *apiv2.GlobalNetworkPolicy) {
+	name := policy.GetObjectMeta().GetName()
+	if !strings.HasPrefix(name, "knp.") {
+		parts := strings.SplitN(name, ".", 2)
+		resetGlobalPolicyName(policy, parts[1])
+	}
+}
+
+func resetGlobalPolicyName(res *apiv2.GlobalNetworkPolicy, originalName string) {
+	res.GetObjectMeta().SetName(originalName)
```

Comment:
Not sure what the point of this function is.

Reply:
Since we are internally prefixing our name data, this prevents the external output from reflecting the internal prefixing that we are adding.

Reply:
Actually I see it's because you needed a function to wrap this up for a defer.